test/extension/config/default/rolebinding.yaml

```
@@ -1,12 +1,13 @@
+# Note: this is specific of the CAPI test-extension, because it uses a ConfigMap to define life-cycle hooks answers.
+# other Runtime extensions provider might want to drop this ClusterRoleBinding or make it scoped to a namespace changing kind to RoleBinding.
 apiVersion: rbac.authorization.k8s.io/v1
-kind: RoleBinding
+kind: ClusterRoleBinding
 metadata:
-  name: test-extension
+  name: manager-rolebinding
 roleRef:
   apiGroup: rbac.authorization.k8s.io
-  kind: Role
-  name: test-extension
+  kind: ClusterRole
+  name: manager-role
 subjects:
   - kind: ServiceAccount
-    name: test-extension
-    namespace: ${TEST_EXTENSION_SERVICE_NAMESPACE}
\ No newline at end of file
+    name: manager
```

```suggestion
    name: manager
    namespace: system
```
in the core provider

---

Not sure why namespace: system is there when it always get overwritten by kustomize...
But overall it fall into similar considerations of https://github.com/kubernetes-sigs/cluster-api/pull/7288#discussion_r981587836 (clusterctl contract is not prescriptive on detail)